src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -16,15 +16,16 @@ public class AuthenticationServiceImpl implements AuthenticationService {
 
     @Override
     public User login(String email, String password) throws AuthenticationException {
-        User user = userService.findByEmail(email).orElseThrow(()
-                -> new AuthenticationException("Can't authenticate user with email " + email
-                + ". There isn't in DB"));
-        String hashedPassword = HashUtil.hashPassword(password, user.getSalt());
-        if (!user.getPassword().equals(hashedPassword)) {
-            throw new AuthenticationException("Can't authenticate user with email, " + email
-                    + ". The password is incorrect ");
+        try {
```

Comment:
It's no a clear handling of the possible AuthenticationException, because you catch all exceptions which can give you false-positive result. Use your previous version but simply add in the `if` check `user.isEmpty()` and give it a general message.